### ISSUE TITLE
DepthwiseConvolutionLayer does not use optimised 3x3 convolutions despite opt3x3=true

### ISSUE THREAD

**Issue by psyhtest** [ARM-software/ComputeLibrary#323]
While profiling the MobileNet graph example using [dvdt-prof](https://github.com/dividiti/dvdt-prof) under [Collective Knowledge](https://github.com/ctuning/ck), I've noticed that despite `DepthwiseConvolutionLayer` being initialised as follows:
```
    SubGraph    sg;
    sg << DepthwiseConvolutionLayer(
           3U, 3U,
           get_weights_accessor(data_path, total_path + "depthwise_depthwise_weights.npy"),
           std::unique_ptr<arm_compute::graph::ITensorAccessor>(nullptr),
           dwc_pad_stride_info,
           true)
```
(where the last argument `opt3x3=true` [means](https://github.com/ARM-software/ComputeLibrary/blob/master/arm_compute/graph/nodes/DepthwiseConvolutionLayer.h#L52) using `DepthwiseConvolutionLayer3x3`), the "direct" `depthwise_convolution_3x3` kernel is not used. Instead, a significantly less optimised sequence of kernels is called including `depthwise_im2col`, `depthwise_weights_reshape`, `depthwise_vector_to_tensor` and `gemm_mv`. Could you confirm please?

**Comment by AnthonyARM** [comment]
Hi Anton,
We'll have a look but it seems to be the same issue as #322

**Comment by psyhtest** [comment]
Thanks @AnthonyARM! Yes, it's similar to #322 "semantically", although quite different from it "syntactically".

**Comment by mohamedadaly** [comment]
Was this issue resolved? I am trying to compare ACL's graph_mobilenet to Tensorflow Lite, and I can see that ACL is much slower. For an image of 224x224 and using 4 threads: ACL takes about 1.3 secs white Tensorflow Lite takes about 0.5 sec. This is on an ARMv7 processor.

Do you have an intuition why ACL is much slower?

**Comment by AnthonyARM** [comment]
What platform is this on ?
How did you time the network ? How many iterations did you do ?

**Comment by mohamedadaly** [comment]
- I am running on a custom board based on Qualcomm's APQ8009 SoC.
- Timed the network like so:
```C
        // Run graph
        for (int i = 0; i < 5; ++i) {
            auto t1 = std::chrono::steady_clock::now();
            graph.run();
            auto t2 = std::chrono::steady_clock::now();
            std::cout << "Took:  " << std::chrono::duration_cast<std::chrono::microseconds>( \
                                  (t2) - (t1)).count() / 1000. << " msec.\n";
        }
```
- I tried 5 iterations, the first is always slower, taking about 1.4 secs, and the rest take about 1.3 secs.

If you could suggest a better way to benchmark the network, that would be great.

**Comment by mohamedadaly** [comment]
I tried to hook up the optimized 3x3 Depthwise Conv layer, and that brought down the run time to about 610 msecs. Compared to TF Lite, the Conv layers in ACL are slower overall, while the Depthwise Conv layers in ACL are faster overall. 

Here is the breakdown I get:

ACL:

Op | Filter | Input | Output | MFLOP | Time
-- | -- | -- | -- | -- | --
Conv: | filter=32x3x3x3 | input=1x224x224x3 | output=1x112x112x32 | 21.676 MFLOP | 25.253 msecs.
Conv: | filter=64x1x1x32 | input=1x112x112x32 | output=1x112x112x64 | 51.3802 MFLOP | 43.042 msecs.
Conv: | filter=128x1x1x64 | input=1x56x56x64 | output=1x56x56x128 | 51.3802 MFLOP | 25.755 msecs.
Conv: | filter=128x1x1x128 | input=1x56x56x128 | output=1x56x56x128 | 102.76 MFLOP | 39.558 msecs.
Conv: | filter=256x1x1x128 | input=1x28x28x128 | output=1x28x28x256 | 51.3802 MFLOP | 22.442 msecs.
Conv: | filter=256x1x1x256 | input=1x28x28x256 | output=1x28x28x256 | 102.76 MFLOP | 28.186 msecs.
Conv: | filter=512x1x1x256 | input=1x14x14x256 | output=1x14x14x512 | 51.3802 MFLOP | 16.113 msecs.
Conv: | filter=512x1x1x512 | input=1x14x14x512 | output=1x14x14x512 | 102.76 MFLOP | 26.801 msecs.
Conv: | filter=512x1x1x512 | input=1x14x14x512 | output=1x14x14x512 | 102.76 MFLOP | 27.092 msecs.
Conv: | filter=512x1x1x512 | input=1x14x14x512 | output=1x14x14x512 | 102.76 MFLOP | 29.369 msecs.
Conv: | filter=512x1x1x512 | input=1x14x14x512 | output=1x14x14x512 | 102.76 MFLOP | 33.376 msecs.
Conv: | filter=512x1x1x512 | input=1x14x14x512 | output=1x14x14x512 | 102.76 MFLOP | 26.512 msecs.
Conv: | filter=1024x1x1x512 | input=1x7x7x512 | output=1x7x7x1024 | 51.3802 MFLOP | 22.305 msecs.
Conv: | filter=1024x1x1x1024 | input=1x7x7x1024 | output=1x7x7x1024 | 102.76 MFLOP | 42.439 msecs.
Conv: | filter=1001x1x1x1024 | input=1x1x1x1024 | output=1x1x1x1001 | 2.05005 MFLOP | 14.97 msecs.
Depthwise Conv: | filter=1x3x3x32 | input=1x112x112x32 | output=1x112x112x32 | 7.22534 MFLOP | 16.71 msecs.
Depthwise Conv: | filter=1x3x3x64 | input=1x112x112x64 | output=1x56x56x64 | 3.61267 MFLOP | 13.446 msecs.
Depthwise Conv: | filter=1x3x3x128 | input=1x56x56x128 | output=1x56x56x128 | 7.22534 MFLOP | 11.609 msecs.
Depthwise Conv: | filter=1x3x3x128 | input=1x56x56x128 | output=1x28x28x128 | 1.80634 MFLOP | 7.517 msecs.
Depthwise Conv: | filter=1x3x3x256 | input=1x28x28x256 | output=1x28x28x256 | 3.61267 MFLOP | 7.127 msecs.
Depthwise Conv: | filter=1x3x3x256 | input=1x28x28x256 | output=1x14x14x256 | 0.903168 MFLOP | 4.009 msecs.
Depthwise Conv: | filter=1x3x3x512 | input=1x14x14x512 | output=1x14x14x512 | 1.80634 MFLOP | 3.636 msecs.
Depthwise Conv: | filter=1x3x3x512 | input=1x14x14x512 | output=1x14x14x512 | 1.80634 MFLOP | 3.813 msecs.
Depthwise Conv: | filter=1x3x3x512 | input=1x14x14x512 | output=1x14x14x512 | 1.80634 MFLOP | 3.693 msecs.
Depthwise Conv: | filter=1x3x3x512 | input=1x14x14x512 | output=1x14x14x512 | 1.80634 MFLOP | 3.643 msecs.
Depthwise Conv: | filter=1x3x3x512 | input=1x14x14x512 | output=1x14x14x512 | 1.80634 MFLOP | 3.857 msecs.
Depthwise Conv: | filter=1x3x3x512 | input=1x14x14x512 | output=1x7x7x512 | 0.451584 MFLOP | 2.263 msecs.
Depthwise Conv: | filter=1x3x3x1024 | input=1x7x7x1024 | output=1x7x7x1024 | 0.903168 MFLOP | 2.616 msecs.

and this is TF Lite:

Op | Filter | Input | Output | MFLOP | Time
-- | -- | -- | -- | -- | --
Conv | filter=32x3x3x3 | input=1x224x224x3 | output=1x112x112x32 | 21.676 MFLOP | 22.417 msecs.
Conv | filter=64x1x1x32 | input=1x112x112x32 | output=1x112x112x64 | 51.3802 MFLOP | 24.68 msecs.
Conv | filter=128x1x1x64 | input=1x56x56x64 | output=1x56x56x128 | 51.3802 MFLOP | 20.373 msecs.
Conv | filter=128x1x1x128 | input=1x56x56x128 | output=1x56x56x128 | 102.76 MFLOP | 31.129 msecs.
Conv | filter=256x1x1x128 | input=1x28x28x128 | output=1x28x28x256 | 51.3802 MFLOP | 14.58 msecs.
Conv | filter=256x1x1x256 | input=1x28x28x256 | output=1x28x28x256 | 102.76 MFLOP | 32.071 msecs.
Conv | filter=512x1x1x256 | input=1x14x14x256 | output=1x14x14x512 | 51.3802 MFLOP | 16.61 msecs.
Conv | filter=512x1x1x512 | input=1x14x14x512 | output=1x14x14x512 | 102.76 MFLOP | 31.835 msecs.
Conv | filter=512x1x1x512 | input=1x14x14x512 | output=1x14x14x512 | 102.76 MFLOP | 34.376 msecs.
Conv | filter=512x1x1x512 | input=1x14x14x512 | output=1x14x14x512 | 102.76 MFLOP | 31.033 msecs.
Conv | filter=512x1x1x512 | input=1x14x14x512 | output=1x14x14x512 | 102.76 MFLOP | 28.786 msecs.
Conv | filter=512x1x1x512 | input=1x14x14x512 | output=1x14x14x512 | 102.76 MFLOP | 27.888 msecs.
Conv | filter=1024x1x1x512 | input=1x7x7x512 | output=1x7x7x1024 | 51.3802 MFLOP | 18.445 msecs.
Conv | filter=1024x1x1x1024 | input=1x7x7x1024 | output=1x7x7x1024 | 102.76 MFLOP | 30.829 msecs.
Conv | filter=1001x1x1x1024 | input=1x1x1x1024 | output=1x1x1x1001 | 2.05005 MFLOP | 8.373 msecs.
DepthwiseConv | filter=1x3x3x32 | input=1x112x112x32 | output=1x112x112x32 | 7.22534 MFLOP | 19.548 msecs.
DepthwiseConv | filter=1x3x3x64 | input=1x112x112x64 | output=1x56x56x64 | 3.61267 MFLOP | 10.992 msecs.
DepthwiseConv | filter=1x3x3x128 | input=1x56x56x128 | output=1x56x56x128 | 7.22534 MFLOP | 18.686 msecs.
DepthwiseConv | filter=1x3x3x128 | input=1x56x56x128 | output=1x28x28x128 | 1.80634 MFLOP | 5.549 msecs.
DepthwiseConv | filter=1x3x3x256 | input=1x28x28x256 | output=1x28x28x256 | 3.61267 MFLOP | 9.165 msecs.
DepthwiseConv | filter=1x3x3x256 | input=1x28x28x256 | output=1x14x14x256 | 0.903168 MFLOP | 2.759 msecs.
DepthwiseConv | filter=1x3x3x512 | input=1x14x14x512 | output=1x14x14x512 | 1.80634 MFLOP | 4.508 msecs.
DepthwiseConv | filter=1x3x3x512 | input=1x14x14x512 | output=1x14x14x512 | 1.80634 MFLOP | 4.142 msecs.
DepthwiseConv | filter=1x3x3x512 | input=1x14x14x512 | output=1x14x14x512 | 1.80634 MFLOP | 4.386 msecs.
DepthwiseConv | filter=1x3x3x512 | input=1x14x14x512 | output=1x14x14x512 | 1.80634 MFLOP | 4.511 msecs.
DepthwiseConv | filter=1x3x3x512 | input=1x14x14x512 | output=1x14x14x512 | 1.80634 MFLOP | 4.15 msecs.
DepthwiseConv | filter=1x3x3x512 | input=1x14x14x512 | output=1x7x7x512 | 0.451584 MFLOP | 1.371 msecs.
DepthwiseConv | filter=1x3x3x1024 | input=1x7x7x1024 | output=1x7x7x1024 | 0.903168 MFLOP | 2.204 msecs.

**Comment by AnthonyARM** [comment]
I confirm this will be fixed in the next release (which should be out by the end of the month)

**Comment by GeorgeARM** [comment]
Resolved in 18.02. Reopen if needed
